pkg/k8s/policyhelper/policy.go

```diff
@@ -20,7 +22,11 @@ type policyInfo struct {
 	kind       gwv1beta1.Kind
 }
 
-func GetAttachedPolicies[T core.Policy](ctx context.Context, k8sClient client.Client, searchTargetRef types.NamespacedName, policy T) ([]T, error) {
+type PolicyConditionFilter struct {
+	Reasons []gwv1alpha2.PolicyConditionReason
+}
+
+func GetAttachedPoliciesConditionFilter[T core.Policy](ctx context.Context, k8sClient client.Client, searchTargetRef types.NamespacedName, policy T, filter PolicyConditionFilter) ([]T, error) {
```

Comment:
Seems this function actually do:  `GetAttachedPoliciesFilterByAcceptedConditionReason` , would be good if we can add comments or change function name to make people more easily know this function's behaviour.

Reply:
removed and refactored for conflict resolution logic